src/main/java/org/vaadin/addons/tatu/TabSheet.java

```diff
@@ -239,20 +249,58 @@ public void setSelected(String tab) {
         getElement().executeJs("this.setSelectedTab($0)", tab);
     }
 
+    /**
+     * Get selected tab index.
+     *
+     * @return Index of the tab, base 0 or -1 of no selection.
+     */
+    public int getSelectedIndex() {
+        return this.getElement().getProperty("selected", -1);
+    }
     /**
      * Set selected tab using index. This will fire TabChangeEvent. Sheet
      * attached to the tab will be shown.
      * 
      * @param index
      *            Index of the tab, base 0.
      */
-    public void setSelected(int index) {
+    public void setSelectedIndex(int index) {
         if (index < 0) {
             throw new IllegalArgumentException("Index can't be negative");
         }
         getElement().setProperty("selected", index);
     }
 
+    /**
+     * Get selected tab Component.
+     *
+     * @return Selected tab Component or <code>null</code> if no selection.
+     */
+    public Component getSelectedTab() {
```

Comment:
getSelectedByComponent ?

Reply:
It returns the Component so "getSelectedComponent" sounds natural alternative. (For me the "by" sounds like the getter would need a Component parameter.)

Reply:
> getSelectedComponent

Yes, it is better actually.

Reply:
Changed to getSelectedComponent